### PR TITLE
build!: update to latest version of gts/typescript

### PR DESCRIPTION
**Pull request by bcoe**
BREAKING CHANGE: typescript@3.7 introduced some breaking changes

### CODE REVIEW

File: samples/package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "engines": {
     "node": ">=8"
   },
+  "files": [],
```
Comment (JustinBeckwith) [inline samples/package.json]: That would effectively disable multiple eslint rules.  Can we make this represent the list of actual files that are used at runtime?

File: src/index.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable prefer-rest-params */
```
Comment (JustinBeckwith) [inline src/index.ts]: We should move this after the license header I think

File: src/index.ts [hunk 3]
```diff
@@ -66,8 +68,7 @@ export function promisify(
   const slice = Array.prototype.slice;
 
   // tslint:disable-next-line:no-any
```
Comment (JustinBeckwith) [inline src/index.ts]: Sadly these won't work with eslint.  There's a different syntax (which I can't remember right now) for disabling a line.

File: src/index.ts (hunk 2 above)
Comment (bcoe) [reply src/index.ts]: removed the eslint-disable inline statements, we will probably want a replacement soonish depending on how migrating other libraries goes.

File: src/index.ts (hunk 3 above)
Comment (AVaksman) [reply src/index.ts]: Screen Shot 2020-03-23 at 1 47 33 PM

Is this useful 👆

**Pull request merged**